src/lib/cpd/CpdEngine.ts

```diff
@@ -0,0 +1,246 @@
+import {Logger, Messages, SfdxError} from '@salesforce/core';
+import {xml2js} from 'xml-js';
+import checksum = require('checksum')
+import {Controller} from '../../Controller';
+import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, TargetPattern} from '../../types';
+import {AbstractRuleEngine} from '../services/RuleEngine';
+import {Config} from '../util/Config';
+import {ENGINE, Severity} from '../../Constants';
+import * as engineUtils from '../util/CommonEngineUtils';
+import CpdWrapper from './CpdWrapper';
+import {FILE_EXTS_TO_LANGUAGE} from '../../Constants';
+import { uxEvents } from '../ScannerEvents';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'CpdEngine');
+
+export class CpdEngine extends AbstractRuleEngine {
+
+	public static ENGINE_ENUM: ENGINE = ENGINE.CPD;
+	public static ENGINE_NAME: string = ENGINE.CPD.valueOf();
```

Comment:
Could you mark these `readonly`? We need to eventually mark the same definitions in other engines as `readonly` as well.